### PR TITLE
CLOUDSTACK-8487 : Add vMotion related tests

### PR DESCRIPTION
**Pull request by abhinavroy02**
Adding a new test script testpath_vMotion_vmware.py in the test/integration/testpath folder. 
This script has vMotion related test cases for VMware.
Tests include :
------------------------
1. Migrate VM with volume within/across the cluster both for vmfs and nfs datastores, windows and linux vms.
2. Migrate VM with volume within/across cluster for local storage.
3. Migrate across cwps and zwps.
4. Migrate across nfs and vmfs.
5. Negative scenarios
6. Migration tests when host is put in maintenance.
7. Migration tests when storage is put in maintenance.
Have also added 2 new functions in base.py

This scripts have been tested on my local setup.
Also, I have run BVTs with this code on my local setup and it doesn't break anything.

### CODE REVIEW

File: tools/marvin/marvin/lib/base.py [hunk 1]
```diff
@@ -2520,6 +2520,35 @@ def reconnect(cls, apiclient, **kwargs):
...
+        returnValue = [FAIL, "VM state not trasited to %s,\
+                        operation timed out" % state]
+
+        while timeout > 0:
+            try:
+                hosts = Host.list(apiclient, 
+                          id=hostid, listall=True)
+                validationresult = validateList(hosts)
+                if validationresult[0] == FAIL:
+                    raise Exception("Host list validation failed: %s" % validationresult[2])
```
Comment (gauravaradhye) [inline tools/marvin/marvin/lib/base.py]: Fix pep8 issues

**Comment by gauravaradhye** [comment]
Don't know why I can't comment on the testpath file. I don't see the file contents in the changes. Any idea why?

**Comment by abhinavroy02** [comment]
Hi Gaurav,

Even I am not getting that option. Is it because it is a new file or
something?

Thanks,
Abhinav

**Comment by gauravaradhye** [comment]
Posting review here as I am not able to put inline review comments.
Comments in the format (line number space comment)


133 space between self,vm

147 There should not be space between parameter=value

133 Function name should resemble an order given to function, like doThis, doThat. change name to createVmSnapshotForDataIntegrityCheck or something suitable

133 Are we not actually checking integrity or only creating Vm snapshot?

164 space between parameters. You can do this automatically using autopep8 tool. pip install autopep8. autopep8 -i -a -a fileName.py. And then check if any pep8 issues are not fixed automatically by pep8 fileName.py and fix them manually.

2480 If test restarts management server by default, then move it to maint folder. Or you can use key "restartManagementServerThroughTestCase"

912 use self.skipTest("") instead of raise unittest.SkipTest, change all such instances

1235 Check if destinationHost is returned as None, in that migrate should not be performed, change everywhere applicable

1278 Compare string after converting to lowercase only. Also don't use string directly, make use of constant strings from codes.py file, change everywhere applicable



Also, post the marvin run logs showing test cases passed successfully.

**Comment by abhinavroy02** [comment]
@gauravaradhye Incorporated the review comments. Please check.

**Comment by abhinavroy02** [comment]
Attaching test execution logs 

This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_1___VMFS____within_cluster____windows__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_2___VMFS____within_cluster____windows__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_3___VMFS____across_cluster____linux__ | Status : SUCCESS ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_4___VMFS____across_cluster____windows__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_5___NetworkFilesystem____within_cluster____linux__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_6___NetworkFilesystem____within_cluster____windows__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_7___NetworkFilesystem____across_cluster____linux__ | Status : SUCCESS ===
ok
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... === TestName: test_01_vm_and_volumes_live_migration_for_vmware_vmfs_8___NetworkFilesystem____across_cluster____windows__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion for VM in local storage within cluster, ... SKIP: The setup doesn't have local storage enabled
This Test Path tests vMotion for VM in local storage within cluster, ... SKIP: The setup doesn't have local storage enabled
This Test Path tests vMotion for VM in local storage within cluster, ... SKIP: The setup doesn't have local storage enabled
This Test Path tests vMotion for VM in local storage within cluster, ... SKIP: The setup doesn't have local storage enabled
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of VMFS type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of VMFS type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of VMFS type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of VMFS type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of NetworkFilesystem type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of NetworkFilesystem type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of NetworkFilesystem type, we need atleast 2
This Test Path tests vMotion for NFS as well as VMFS within cluster, ... SKIP: The setup doesn't have enough zone wide primary storages of NetworkFilesystem type, we need atleast 2
This Test Path tests vMotion from NFS ok---> VMFS , within cluster ... === TestName: test_04_vm_and_volumes_live_migration_for_vmware_across_nfs_vmfs_1___within_cluster____linux__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion from NFS ok---> VMFS , within cluster ... === TestName: test_04_vm_and_volumes_live_migration_for_vmware_across_nfs_vmfs_2___within_cluster____windows__ | Status : EXCEPTION ===
ok
This Test Path tests vMotion from NFS ok---> VMFS , within cluster ... SKIP: The setup doesn't have enough pools or enough hosts. To run these tests the setup must have atleast 2 clusters, each having min 2 host 2 vmfs storage pools and 2 nfs storage pools
This Test Path tests vMotion from NFS ok---> VMFS , within cluster ... SKIP: The setup doesn't have enough pools or enough hosts. To run these tests the setup must have atleast 2 clusters, each having min 2 host 2 vmfs storage pools and 2 nfs storage pools
Test scenarios like : ... === TestName: test_05_vm_and_volumes_live_migration_for_vmware_negative_scenarios | Status : SUCCESS ===
ok
Test scenarios for Host Maintenance ... === TestName: test_06_vm_and_volumes_live_migration_for_vmware_host_maintenance | Status : EXCEPTION ===
ok
Test scenarios for storage Maintenance ... === TestName: test_07_vm_and_volumes_live_migration_for_vmware_storage_maintenance | Status : EXCEPTION ===
ok

**Comment by gauravaradhye** [comment]
This looks good and all review comments have been incorporated. Merging this.

**Pull request closed without merging**